src/store/slices/toast/toastSlice.types.ts

```diff
@@ -0,0 +1,82 @@
+import { JSX } from 'react';
+import { ServerTransactionType } from 'types/serverTransactions.types';
+import { SignedTransactionType } from 'types/transactions.types';
+
+export interface ToastsSliceState {
+  customToasts: CustomToastType[];
+  transactionToasts: TransactionToastType[];
+}
+
+interface SharedCustomToast {
+  toastId: string;
+  /**
+   * Duration in miliseconds
+   */
+  duration?: number;
+  type?: string;
+  onClose?: () => void;
+}
+export interface MessageCustomToastType extends SharedCustomToast {
+  message: string;
+  icon?: never;
+  iconClassName?: never;
+  title?: never;
+  status?: never;
+  transaction?: never;
+  component?: never;
+}
+interface SharedIconToastType extends SharedCustomToast {
+  icon: string;
+  iconClassName?: string;
+  title: string;
+}
+export interface MessageIconToastType extends SharedIconToastType {
+  message: string;
+  /**
+   * Use `status` to display a row of information between `title` and `message`
+   */
+  status?: string;
+  transaction?: never;
+  component?: never;
+}
+
+export interface TransactionIconToastType extends SharedIconToastType {
+  transaction: ServerTransactionType;
+  component?: never;
+  message?: never;
+  status?: never;
+}
+
+export interface ComponentIconToastType extends SharedIconToastType {
+  /**
+   * Use `component` to display a custom React compnent
+   *
+   * **⚠️ Warning**: Toasts with components will not be persisted on page reload because React components are not serializable
+   */
+  component: (() => JSX.Element) | null;
```

Comment:
is this used? since we are no longer in React

Reply:
i removed for now all the custom types and let only the generic one